internal/server/handler_v2.go

```diff
@@ -18,14 +18,57 @@ package server
 import (
 	"context"
 
+	v2 "github.com/datacommonsorg/mixer/internal/server/v2"
 	v2pv "github.com/datacommonsorg/mixer/internal/server/v2/propertyvalues"
+	"github.com/datacommonsorg/mixer/internal/util"
 
 	pb "github.com/datacommonsorg/mixer/internal/proto"
 )
 
-// PropertyValuesV2 implements API for mixer.PropertyValuesV2.
-func (s *Server) PropertyValuesV2(
-	ctx context.Context, in *pb.PropertyValuesV2Request,
-) (*pb.PropertyValuesV2Response, error) {
-	return v2pv.API(ctx, in, s.store)
+// QueryV2 implements API for mixer.QueryV2.
+func (s *Server) QueryV2(
+	ctx context.Context, in *pb.QueryV2Request,
+) (*pb.QueryV2Response, error) {
+	arcStrings, err := v2.SplitArc(in.Graph)
+	if err != nil {
+		return nil, err
+	}
+	arcs := []*v2.Arc{}
+	for _, s := range arcStrings {
+		arc, err := v2.ParseArc(s)
+		if err != nil {
+			return nil, err
+		}
+		arcs = append(arcs, arc)
+	}
+	// Simple Property Values
+	if len(arcs) == 1 {
```

Comment:
I assume you'll add the case "len(arcs) != 1" in a future PR. So the logic here becomes pretty long, can we abstract out to a function? Ideally hander.go only does routing and does not contain complex logic?